src/data/repositories/utils/DiseaseOutbreakMapper.ts

```diff
@@ -0,0 +1,252 @@
+                    "true",
+            },
+            initiateRiskCommunication: {
+                date: new Date(getValueFromMap("initiateRiskCommunicationDate", trackedEntity)),
+                na: getValueFromMap("initiateRiskCommunicationNA", trackedEntity) === "true",
+            },
+            establishCoordination: new Date(
+                getValueFromMap("establishCoordination", trackedEntity)
+            ),
+            responseNarrative: getValueFromMap("responseNarrative", trackedEntity),
+        },
+        notes: getValueFromMap("notes", trackedEntity),
+    };
+
+    return diseaseOutbreak;
+}
+
+export function mapDiseaseOutbreakEventToTrackedEntityAttributes(
+    diseaseOutbreak: DiseaseOutbreakEventBaseAttrs,
+    attributesMetadata: D2TrackedEntityAttribute[]
+): D2TrackerTrackedEntity {
+    const attributes: Attribute[] = attributesMetadata.map(attribute => {
+        const populatedAttribute = {
+            attribute: attribute.trackedEntityAttribute.id,
+            value: getValueFromDiseaseOutbreak(
+                attribute.trackedEntityAttribute
+                    .code as (typeof DiseaseOutbreakCodes)[keyof typeof DiseaseOutbreakCodes], //TO DO :  Can we avoid?
+                diseaseOutbreak
+            ),
+        };
+        return populatedAttribute;
+    });
+    const enrollment: D2TrackerEnrollment = {
+        orgUnit: RTSL_ZEBRA_ORG_UNIT_ID,
+        program: RTSL_ZEBRA_PROGRAM_ID,
+        enrollment: "",
+        trackedEntityType: RTSL_ZEBRA_TRACKED_ENTITY_TYPE_ID,
+        notes: [],
+        relationships: [],
+        attributes: attributes,
+        events: [],
+        enrolledAt: diseaseOutbreak.created.toISOString(),
+        occurredAt: diseaseOutbreak.lastUpdated.toISOString(),
+        createdAt: new Date().getTime().toString(),
+        createdAtClient: new Date().getTime().toString(),
+        updatedAt: new Date().getTime().toString(),
+        updatedAtClient: new Date().getTime().toString(),
+        status: "ACTIVE",
+        orgUnitName: "",
+        followUp: false,
+        deleted: false,
+        storedBy: "",
+    };
+    const trackedEntity: D2TrackerTrackedEntity = {
+        trackedEntity: diseaseOutbreak.id,
+        orgUnit: RTSL_ZEBRA_ORG_UNIT_ID,
+        trackedEntityType: RTSL_ZEBRA_TRACKED_ENTITY_TYPE_ID,
+        createdAt: diseaseOutbreak.created.toISOString(),
+        updatedAt: diseaseOutbreak.lastUpdated.toISOString(),
+        attributes: attributes,
+        enrollments: [enrollment],
+    };
+
+    return trackedEntity;
+}
+
+function getValueFromMap(
+    key: keyof typeof DiseaseOutbreakCodes,
+    trackedEntity: D2TrackerTrackedEntity
+): string {
+    return trackedEntity.attributes?.find(a => a.code === DiseaseOutbreakCodes[key])?.value ?? "";
+}
+
+function getValueFromDiseaseOutbreak(
+    key: (typeof DiseaseOutbreakCodes)[keyof typeof DiseaseOutbreakCodes],
+    diseaseOutbreak: DiseaseOutbreakEventBaseAttrs
+): string {
```

Comment:
I can't see a drawback of the mapping approach, but it left me thinking what would be the non-mapping approach :) I think it would be something like this:

```ts
const attributeValues = {
  "RTSL_ZEB_TEA_EVENT_id": diseaseOutbreak.eventId.toString(),
  "RTSL_ZEB_TEA_EVENT_NAME": diseaseOutbreak.name,
  // ...
}
const attributes =getAttributesFromAttributeValues(attributeValues)
```

With both styles at hand, pick the one that looks more declarative to you.